openid-zkp-auth/circuits/jwt_proof_ua.circom

```diff
@@ -0,0 +1,170 @@
+pragma circom 2.1.3;
+
+include "helpers/sha256.circom";
+include "helpers/misc.circom";
+include "helpers/strings.circom";
+include "helpers/zkhasher.circom";
+
+/**
+JWT Proof: User-agnostic (UA) circuit
+
+    Construction params:
+    - maxContentLength:         Maximum length of the JWT + SHA2 padding in bytes
+    - maxSubLength:             Maximum length of the subject_id (in ascii)
+
+    Private Inputs:
+    - content[inCount]:         Segments of X as inWidth bit chunks where X is JWT header + JWT payload + SHA-2 padding + zeroes
+    - sub_length_ascii:         Length of the subject_id in ASCII, e.g., for ',"sub":12345,' it is 13
+    - subject_id[maxSubLength]: The subject (user) ID for the first sub_length_ascii characters and 0s for the rest
+    - subject_pin:              User's PIN to keep the subject_id private
+    - mask[inCount]:            A binary mask over X, i.e., mask[i] = 0 or 1
+    - randomness:               A 248-bit random number to keep the sensitive parts of JWT hidden
+
+    Circuit signals revealed to the verifier along with the ZK proof:
+    - jwt_sha2_hash:            The SHA2 hash of the JWT header + JWT payload + SHA-2 padding
+    - num_sha2_blocks:          Number of SHA2 (64-byte) blocks the SHA2-padded JWT consumes
+    - subject_id_com:           H(subject_id || PIN). A binding and hiding commitment to subject_id
+    - payload_start_index:      The index of the payload in the content
+    - payload_len:              The length of the payload
+    - masked_content:           The content with the sensitive parts masked
```

Comment:
please list the fields that are expected to be revealed

Reply:
Added.